parcels/particleset.py

```diff
@@ -238,38 +254,113 @@ def execute(self, pyfunc=AdvectionRK4, starttime=None, endtime=None, dt=1.,
-            namestr = ''
-            time_origin = self.grid.U.time_origin
+        t = kwargs.get('t', time[0])
+        if isinstance(t, datetime):
+            t = (t - self.grid.U.time_origin).total_seconds()
+        if isinstance(t, delta):
+            t = t.total_seconds()
+        if domain is not None:
+            latN = nearest_index(self.grid.U.lat, domain[0])
+            latS = nearest_index(self.grid.U.lat, domain[1])
+            lonE = nearest_index(self.grid.U.lon, domain[2])
+            lonW = nearest_index(self.grid.U.lon, domain[3])
+        else:
+            latN, latS, lonE, lonW = (-1, 0, 0, -1)
+        if not vector:
+            t = int(t)
+            if plt is None:
+                raise RuntimeError("Visualisation not possible: matplotlib not found!")
+            field = kwargs.get('field', True)
+            plt.ion()
+            plt.clf()
+            plt.plot(np.transpose(plon), np.transpose(plat), 'ko')
+            if field is True:
+                axes = plt.gca()
+                axes.set_xlim([self.grid.U.lon[lonE], self.grid.U.lon[lonW]])
+                axes.set_ylim([self.grid.U.lat[latS], self.grid.U.lat[latN]])
+                namestr = ''
+                time_origin = self.grid.U.time_origin
+            else:
+                if not isinstance(field, Field):
+                    field = getattr(self.grid, field)
+                field.show(with_particles=True, **dict(kwargs, t=t))
+                namestr = ' on ' + field.name
+                time_origin = field.time_origin
+            if time_origin is 0:
+                timestr = ' after ' + str(delta(seconds=t)) + ' hours'
+            else:
+                timestr = ' on ' + str(time_origin + delta(seconds=t))
+            plt.xlabel('Longitude')
+            plt.ylabel('Latitude')
+            plt.title('Particles' + namestr + timestr)
         else:
-            if not isinstance(field, Field):
-                field = getattr(self.grid, field)
-            field.show(with_particles=True, **dict(kwargs, t=t))
-            namestr = ' on ' + field.name
-            time_origin = field.time_origin
-        if time_origin is 0:
-            timestr = ' after ' + str(delta(seconds=t)) + ' hours'
+            if Basemap is None:
+                raise RuntimeError("Visualisation not possible: Basemap module not found!")
+            land = kwargs.get('land', False)
+            vmax = kwargs.get('vmax', None)
+            idx = self.grid.U.time_index(t)
+            U = np.array(self.grid.U.temporal_interpolate_fullfield(idx, t))
+            V = np.array(self.grid.V.temporal_interpolate_fullfield(idx, t))
+            lon = self.grid.U.lon
+            lat = self.grid.U.lat
+            if domain is not None:
+                lon = lon[lonW:lonE]
+                lat = lat[latS:latN]
+                U = U[latS:latN, lonW:lonE]
+                V = V[latS:latN, lonW:lonE]
```

Comment:
Given that we set `latN, latS, lonE, lonW = (-1, 0, 0, -1)` on line 274 above, can't this always be executed (i.e. is there need for the `if domain is not None:` statement here)?